src/steps/card-register/index.tsx

```diff
@@ -34,17 +47,58 @@ export const CardInputFormStep = ({ onSubmit, onClickPrev }: CardInputFormStepPr
 
   const handleSubmit: FormEventHandler<HTMLFormElement> = e => {
     e.preventDefault()
-    if (!cardType) return
+    if (!validateCardInput(cardInput)) return
     onSubmit?.({ ...cardInput, cardType })
   }
 
   const handleClickCard = () => {
-    openBottomSheet(
+    openOverlay(
       <CardTypePickBottomSheet
         cardTypeList={Object.values(CARD_TYPE)}
-        onClose={closeBottomSheet}
+        onClose={closeOverlay}
         selectedCardType={cardType}
-        onSelectCardType={setCardInput('cardType')}
+        onSelectCardType={value => {
+          setCardInput('cardType')(value)
+          cardCodeInputRef.current?.focus()
+        }}
+      />,
+    )
+  }
+
+  const handleChangeCardCodeInput = (value: string) => {
+    if (value.length === 8) {
+      setCardInput('cardType')(getCardTypeFromCardCode(value))
+    }
+    setCardInput('cardCode')(value)
+  }
+
+  const handleopenCVCNumberPad = () => {
```

Comment:
c: 케이스 수정 필요한 부분이 있네요
```suggestion
  const handleOpenCVCNumberPad = () => {
```